File: use-cases/easy-ship/code/java/src/main/java/lambda/GetScheduledPackageHandler.java
package lambda;

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.google.gson.Gson;
import io.swagger.client.api.EasyShipApi;
import io.swagger.client.model.easyship.ModelPackage;
import lambda.utils.ApiUtils;
import lambda.utils.StateMachineInput;


public class GetScheduledPackageHandler implements RequestHandler<StateMachineInput, StateMachineInput> {

    public StateMachineInput handleRequest(StateMachineInput input, Context context) {
        LambdaLogger logger = context.getLogger();
        logger.log("getScheduledPackage Lambda input: " + new Gson().toJson(input));

        try {
            EasyShipApi easyShipApi = ApiUtils.getEasyShipApi(input);
            ModelPackage response = easyShipApi.getScheduledPackage(input.getAmazonOrderId(), input.getMarketplaceId());
            logger.log("EasyShip API -  CreateScheduledPackage response: " + new Gson().toJson(response));
            // Validate if scheduled correctly using ScheduledPackageId
            if (!response.getScheduledPackageId().equals(input.getScheduledPackageId())) {
                throw new IllegalArgumentException(
                        String.format("Amazon Order Id : %s was not scheduled correctly", input.getAmazonOrderId()));
            }
            return input;
        } catch (Exception e) {
            throw new InternalError("Calling EasyShip API failed", e);
        }
    }
}

File: use-cases/easy-ship/code/php/lambda/GetScheduledPackageHandler.php
<?php

namespace Lambda;

use GuzzleHttp\Exception\RequestException;
use Lambda\Utils\ApiUtils;
use Lambda\utils\LambdaContext;
use Lambda\Utils\Model\StateMachineInput;
use Lambda\Utils\StateMachineInputConverter;

class GetScheduledPackageHandler
{
    public function handleRequest(array $event, LambdaContext $context): StateMachineInput
    {
        $input = StateMachineInputConverter::convertFromArray($event);
        $logger = $context->getLogger();
        $logger->info('getScheduledPackage Lambda input: ' . json_encode($event, JSON_PRETTY_PRINT));

        try {
            // Initialize EasyShip API client
            $easyShipApi = ApiUtils::getEasyShipApi($input);

            // Call the EasyShip API to get the scheduled package
            $response = $easyShipApi->getScheduledPackage(
                $input->getAmazonOrderId(),
                $input->getMarketplaceId()
            );
            $logger->info('EasyShip API - GetScheduledPackage response: ' . json_encode($response));
            // Validate if the scheduled package ID matches the expected ID
            if (trim($response->getScheduledPackageId()->getPackageId())
                !== trim($input->getScheduledPackageId()->getPackageId())) {
                throw new \InvalidArgumentException(sprintf(
                    'Amazon Order Id: %s was not scheduled correctly',
                    $input->getAmazonOrderId()
                ));
            }

            return $input;
        } catch (RequestException $e) {
            $logger->error('RequestException: ' . $e->getMessage());
            throw new \RuntimeException('Calling EasyShip API failed', 0, $e);
        } catch (\Exception $e) {
            $logger->error('Exception: ' . $e->getMessage());
            throw new \RuntimeException('An unexpected error occurred', 0, $e);
        }
    }
}
